psysh.el

```diff
@@ -188,6 +203,39 @@ See `psysh-mode-output-syntax-table'."
     (psysh)))
 
 
+;; History
+(defun psysh--config-dir-path ()
+  "Return path to PsySH config dir."
+  ;; TODO: maybe next version Emacs bundles xdg.el?
+  (if (eq system-type 'windows-nt)
+      (f-join (s-replace-all '(("\\" . "/"))
+                             (or (getenv "APPDATA")
+                                 (f-join (getenv "HOME") "AppData"))) "PsySH")
+    (f-join (or (getenv "XDG_CONFIG_HOME")
+                (f-join (getenv "HOME") ".config"))
+            "psysh")))
+
+(defun psysh--load-history (path n)
+  "Load input histories by PATH and return N elements."
+  (with-temp-buffer
+    (insert-file-contents-literally path)
+    (goto-char (point-max))
+    (reverse
+     (cl-loop repeat n
+              do (beginning-of-line)
+              never (eq (point) (point-min))
+              collect (buffer-substring-no-properties (point)
+                                                      (save-excursion (end-of-line)
+                                                                      (point)))
+              do (forward-line -1)))))
+
+(defun psysh--insertion-history-lines (histories)
+  "Insert history `HISTORIES' lines to comint-input-ring."
```

Comment:
Same as above, I think it will be enough to then say just `Insert HISTORY lines to ...` and rename the variable accordingly.